src/rule.rs

```diff
@@ -418,21 +418,39 @@ impl IRule {
             Rule::Concat([left, right]) => NodeShape::Split(left, right),
             Rule::Or(ref cases) => NodeShape::Choice(cases.len()),
             Rule::Opt(rule) => NodeShape::Opt(rule),
-            Rule::RepeatMany(elem, sep) => NodeShape::Opt(cx.intern(Rule::RepeatMore(elem, sep))),
-            Rule::RepeatMore(rule, None) => {
-                NodeShape::Split(rule, cx.intern(Rule::RepeatMany(rule, None)))
+            Rule::RepeatMany(..) | Rule::RepeatMore(..) => {
+                NodeShape::Alias(self.expand_repeats(cx))
```

Comment:
This needs to be reverted/replaced with `self.expand_repeats(cx).node_shape(cx, named_rules)`.